mpp/include/mpp_define_nest_domains.inc

```diff
@@ -1240,40 +1518,825 @@ function search_C2F_nest_overlap(nest_domain, extra_halo, position)
+    enddo
+
+
+  end subroutine get_nnest
+
+
+!--- This routine will convert the global coarse grid index to nest grid index.
+  function convert_index_to_nest(domain, ishift, jshift, tile_coarse, istart_coarse, iend_coarse, jstart_coarse, jend_coarse, &
+                       ntiles_coarse, tile_in, is_in, ie_in, js_in, je_in, is_out, ie_out, js_out, je_out, rotate_out)
+    type(domain2D), intent(in) :: domain
+    integer, intent(in)  :: ishift, jshift
+    integer, intent(in)  :: istart_coarse, iend_coarse, jstart_coarse, jend_coarse
+    integer, intent(in)  :: tile_coarse
+    integer, intent(in)  :: ntiles_coarse, tile_in, is_in, ie_in, js_in, je_in
+    integer, intent(out) :: is_out(:), ie_out(:), js_out(:), je_out(:), rotate_out(:)
+    integer              :: convert_index_to_nest
+    integer :: is, ie, js, je, tile, isg, ieg, jsg, jeg
+    integer :: ncross, rotate, nout, diff, l, ntiles
+
+    ntiles = ntiles_coarse
+    call mpp_get_global_domain(domain, isg, ieg, jsg, jeg)
+    is = istart_coarse; ie = iend_coarse
+    js = jstart_coarse; je = jend_coarse
+    tile = tile_coarse
+
+    if(size(is_out(:)) < 2) call mpp_error(FATAL,"convert_index_to_nest: size(is_out(:)) < 2")
+    if(size(ie_out(:)) < 2) call mpp_error(FATAL,"convert_index_to_nest: size(ie_out(:)) < 2")
+    if(size(js_out(:)) < 2) call mpp_error(FATAL,"convert_index_to_nest: size(js_out(:)) < 2")
+    if(size(je_out(:)) < 2) call mpp_error(FATAL,"convert_index_to_nest: size(je_out(:)) < 2")
+    if(size(rotate_out(:)) < 2) call mpp_error(FATAL,"convert_index_to_nest: size(rotate_out(:)) < 2")
+    if( ie > ieg .AND. je > jeg) then
+       call mpp_error(FATAL, "convert_index_to_nest: do not support cross the corner, contact developer")
+    endif
+    if( is > ieg .or. js > jeg) call mpp_error(FATAL,"convert_index_to_nest: is > ieg .or. js > jeg")
+
+
+    nout = 0
+
+    if(tile == tile_in) then
+       nout = nout+1
+       rotate_out(nout) = ZERO
+       is_out(nout) = is_in; ie_out(nout) = ie_in + ishift
+       js_out(nout) = js_in; je_out(nout) = je_in + jshift
+    endif
+
+    diff =  tile_in - tile
+    if(diff < 0) diff = diff + ntiles
+    ncross = -1
+    if( ie > ieg ) then
+       select case(diff)
+       case (0)
+          rotate = ZERO
+          ncross = 4
+       case (1)
+          if(mod(tile,2) ==1) then ! tile 1 3 5
+             rotate = ZERO
+             ncross = 1
+          endif
+       case (2)
+          if(mod(tile,2) ==0) then ! tile 2 4 6
+             rotate = MINUS_NINETY
+             ncross = 1
+          endif
+       case (3)
+          rotate = MINUS_NINETY
+          ncross = 2
+       case (4)
+          if(mod(tile,2) ==1) then ! tile 1 3 5
+             rotate = MINUS_NINETY
+             ncross = 3
+          endif
+       case (5)
+          if(mod(tile,2) ==0) then ! tile 2 4 6
+             rotate = ZERO
+             ncross = 3
+          endif
+       case default
+print*,"diff=", diff, tile_in, tile, ntiles
```

Comment:
Is this a left over debug print that should be removed?

Reply:
It certainly appears as if it can be removed.